src/mgba_printf/mini_printf.c

```diff
@@ -49,13 +49,90 @@
 #include "mgba_printf/mini_printf.h"
 #include "gba/types.h"
 #include "gba/defines.h"
+#include "text.h"
 
 struct mini_buff 
 {
     char *buffer, *pbuffer;
     u32 buffer_len;
 };
 
+//This converts the number used to represent a character in the game code to a regular char (character).
+static u8 decodeChar(u8 input)
```

Comment:
This is incomplete, also referencing each alphanumerical character sounds a bit useless, also there is already an implementation here: <https://github.com/rh-hideout/pokeemerald-expansion/blob/c2d880fce9ba4fbe499da78aff0cb81ded65a257/src/mgba_printf/mgba.c#L46>